docs/usage/dataclasses.md

```diff
@@ -378,25 +378,35 @@ class User:
 
     @model_validator(mode='before')
     def pre_root(cls, values: Dict[str, Any]) -> Dict[str, Any]:
-        print(values)
-        #> ArgsKwargs((), {'birth': {'year': 1995, 'month': 3, 'day': 2}})
+        print(f'First: {values}')
+        """
+        First: ArgsKwargs((), {'birth': {'year': 1995, 'month': 3, 'day': 2}})
+        """
         return values
 
     @model_validator(mode='after')
     def post_root(self) -> 'User':
-        print(self)
-        #> User(birth=Birth(year=1995, month=3, day=2))
+        print(f'Third: {self}')
+        #> Third: User(birth=Birth(year=1995, month=3, day=2))
         return self
 
     def __post_init__(self):
-        print(self.birth)
-        #> Birth(year=1995, month=3, day=2)
+        print(f'Second: {self.birth}')
+        #> Second: Birth(year=1995, month=3, day=2)
 
 
 user = User(**{'birth': {'year': 1995, 'month': 3, 'day': 2}})
 ```
 
-The `__post_init__` in Pydantic dataclasses is called _after_ validation, rather than before.
+The `__post_init__` in Pydantic dataclasses is called in the _middle_ of validators.
+Here is the order:
+
+* Before model validator.
+* Before field validator.
+* After field validator.
+* Field type validators. e.g. validation for types like `int`, `str`, ... .
+* `__post_init__`.
+* After model validator.
```

Comment:
I think `model_validator(mode='before')`, `field_validator(mode='before')`, etc. notation looks better. 👀 

Also, I think we call the "Field type validators" differently in the rest of the documentation? 🤔 Inner validator? I'm not sure...

Reply:
Ok, look better.
I changed them.